cogs/commands/system.py

```diff
@@ -0,0 +1,155 @@
+import logging
+import platform
+from datetime import datetime
+from os import environ
+from typing import Any
+
+import aiohttp
+import discord
+from discord.ext import commands
+from discord.ext.commands import Bot, Context
+from sqlalchemy import select
+
+from config import CONFIG
+from models.models import db_session
+from models.system import EventKind, SystemEvent
+
+
+class System(commands.Cog):
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+        try:
+            with open(".version", "r") as f:
+                content = f.readlines()
+                self.version_from_file = content[0].rstrip()
+                self.build_timestamp_from_file = content[1].rstrip()
+        except FileNotFoundError:
+            logging.error("Could not load git revision from file")
+            self.version_from_file = None
+            self.build_timestamp_from_file = None
+
+    @commands.hybrid_command()
+    async def info(self, ctx: Context[Bot]):
+        json = await self.get_docker_json()
+        if json is None:
+            return
+
+        # these are likely to be outdated, so we don't use them.
+        # version = json["Config"]["Labels"]["org.opencontainers.image.revision"]
+        # built = json["Config"]["Labels"]["org.opencontainers.image.created"]
+
+        version = self.version_from_file
+        built = self.build_timestamp_from_file
+
+        description: str = json["Config"]["Labels"][
+            "org.opencontainers.image.description"
+        ]
+
+        py_version = platform.python_version()
+        dpy_version = discord.__version__
+
+        # the timestamp docker gives does not conform to ISO.
+        # we strip the fractional secconds and add the suffix which assumes UTC (may cause issues in summer)
+        timestamp: str = json["State"]["StartedAt"]
+        started = datetime.fromisoformat(timestamp.split(".")[0])
+        uptime = datetime.now() - started
```

Comment:
If docker time is UTC, best to use `.utcnow()` and localize `started`

Reply:
Interesting
Python 3.10 does not have full ISO 8601 support, hence why this confused me: I assumed python was right.
Python 3.11 has much better support...